### PR TITLE
changed security configuration of server/client

### PR DESCRIPTION
**Pull request by akhon**
- client keys always needed
- auth-user-pass always needed
- google_2fa turns off only ta.key

relates to https://github.com/verygood-ops/infra-vault/issues/187
resolves #71

### CODE REVIEW

**Comment by akhon** [comment]
needs for final testing on staging

**Comment by mjallday** [comment]
@akhon please verify and then we can merge

File: templates/server.conf.j2 [hunk 1]
```diff
@@ -79,12 +79,11 @@ mute 10  # suppress identical messages > 10 occurrences
 reneg-sec 0
 script-security 1
 
-{% if openvpn_2fa_google == 'yes' %}
-client-cert-not-required
```
Comment (mahmoudimus) [inline templates/server.conf.j2]: we do not need a client cert. why do we need one? this is a bug imo

File: templates/client.ovpn.j2 [hunk 2]
```diff
@@ -9,12 +9,12 @@ persist-key
 persist-tun
 
 ca ca.crt
-
-{% if openvpn_2fa_google == 'yes' %}
-auth-user-pass
-{% else %}
 cert {{item}}.crt
```
Comment (mahmoudimus) [inline templates/client.ovpn.j2]: this is not needed

Comment (mahmoudimus) [inline templates/client.ovpn.j2]: if you are using 2fa, which is password and 2fa token, why do you need a client cert? This is wrong

File: tasks/main.yml [hunk 3]
```diff
@@ -37,15 +37,15 @@
...
+  when: clients is defined
   tags:
     - client_keys
 
-- name: generate client configs
-  include: client_keys_2fa.yml
-  when: clients and openvpn_2fa_google == 'yes'
-  tags:
-    - client_keys
+#- name: generate client configs
```
Comment (mahmoudimus) [inline tasks/main.yml]: what was the difference between the two client configs? I don't remember

**Comment by mahmoudimus** [comment]
I disagree with this pull request. 👎👎👎👎👎

File: tasks/main.yml (hunk 3 above)
Comment (akhon) [inline tasks/main.yml]: in generating and fetching client key and cert

**Comment by akhon** [comment]
here's a warning from openvpn verbose (it's about ta.key):
```
Thu Jun  9 15:24:55 2016 WARNING: No server certificate verification method has been enabled.  See http://openvpn.net/howto.html#mitm for more info.
```

**Comment by mahmoudimus** [comment]
@akhon ok :) we can go ahead with that. how will this playbook work when we bootstrap a new openvpn machine via auto scaling?

**Comment by akhon** [comment]
@mahmoudimus, at bootstrap it will install openvpn server with config from template

**Comment by mahmoudimus** [comment]
@akhon that is very unhelpful because "duh, I can already tell you it does that." How will it work for existing users of an openvpn machine that is reprovisioned? Will we need to regenerate new certs, passwords, etc for them or will we have to regenerate EVERYTHING again and give new credentials to each person?

**Comment by mahmoudimus** [comment]
I need you to figure out what our plan for this going forward

**Comment by akhon** [comment]
@mahmoudimus, ah that's a question.. what if we just put all generated keys somewhere (i.e. citadel bucket) and have an option in playbook to restore keys? anyway, this is a part of https://github.com/verygood-ops/infra-vault/issues/71

**Comment by mahmoudimus** [comment]
@akhon can you solve this problem? if you can come up with a solution, I would appreciate it

**Comment by akhon** [comment]
@mahmoudimus, sure!

**Comment by akhon** [comment]
@mahmoudimus, please take a look. we are storing keys by default:
- server keys: `{{citadel_bucket}}/openvpn/{{ansible_dns.search[0]}}/keys/server`
- clients keys: `{{citadel_bucket}}/openvpn/{{ansible_dns.search[0]}}/keys/clients`
- client's configs: `{{citadel_bucket}}/openvpn/{{ansible_dns.search[0]}}/configs`

infra changes are needed, see https://github.com/verygood-ops/infra-vault/pull/195

tested and implemented in staging.

let me know if this is ok, requirements.yml will need to be updated in both environments

/cc @mjallday

**Comment by mjallday** [comment]
@akhon possible to get the tests passing?

**Comment by akhon** [comment]
@mjallday yes, they are not passing because they doesn't exists. it's planned to implement circleci for all oiur playbooks in low-priority task: https://github.com/verygood-ops/infra-processing/issues/167

File: defaults/main.yml [hunk 4]
```diff
@@ -6,6 +6,14 @@ openvpn_service_name: "openvpn"
 openvpn_config_file: "openvpn_{{ openvpn_proto }}_{{ openvpn_port }}"
 cname: "{{ ansible_fqdn }}"
 
+# backup section
+openvpn_backup_keys: true 
+backup_bucket: '{{ citadel_bucket }}'
```
Comment (mjallday) [inline defaults/main.yml]: these variables should be prefixed with `openvpn_` to namespace them. otherwise they can be clobbered by other ansible plays.

Comment (akhon) [inline defaults/main.yml]: added!

**Pull request merged**